client/pool-buffer.c

```diff
@@ -103,30 +103,29 @@ void destroy_buffer(struct pool_buffer *buffer) {
 }
 
 struct pool_buffer *get_next_buffer(struct wl_shm *shm,
-		struct pool_buffer pool[static 2], uint32_t width, uint32_t height) {
-	struct pool_buffer *buffer = NULL;
-
-	for (size_t i = 0; i < 2; ++i) {
-		if (pool[i].busy) {
-			continue;
-		}
-		buffer = &pool[i];
-	}
-
-	if (!buffer) {
-		return NULL;
-	}
-
-	if (buffer->width != width || buffer->height != height) {
-		destroy_buffer(buffer);
-	}
-
-	if (!buffer->buffer) {
-		if (!create_buffer(shm, buffer, width, height,
-					WL_SHM_FORMAT_ARGB8888)) {
-			return NULL;
-		}
-	}
-	buffer->busy = true;
-	return buffer;
+                                    struct pool_buffer pool[static 2],
+                                    uint32_t width, uint32_t height) {
+    for (size_t i = 0; i < 2; ++i) {
+        if (!pool[i].busy &&
+            (pool[i].width == width && pool[i].height == height)) {
+            pool[i].busy = true;
+            return &pool[i];
+        }
+    }
+
+    for (size_t i = 0; i < 2; ++i) {
+        if (!pool[i].busy) {
+            if (pool[i].width != width || pool[i].height != height) {
```

Comment:
This condition is always true, as we would have returned from the function during the first loop if a non-busy buffer had matching dimensions.